Sources/Orbit/Components/SegmentedSwitch.swift

```diff
@@ -0,0 +1,214 @@
+import SwiftUI
+
+/// Showing two choices from which only one can be selected.
+///
+/// Currently SegmentedSwitch allows to have 2 segments.
+/// SegmentedSwitch can be in error state only in unselected state.
+///
+/// - Note: [Orbit definition](https://orbit.kiwi/components/interaction/segmentedswitch/)
+public struct SegmentedSwitch: View {
+
+    @Binding private var selectedIndex: Int?
+    let label: String
+    let message: Message?
+    let firstOption: String
+    let secondOption: String
+
+    let separatorWidth: CGFloat = BorderWidth.emphasis
+    let borderWidth: CGFloat = BorderWidth.emphasis
+
+    public var body: some View {
+        FieldWrapper(
+            label,
+            message: message
+        ) {
+            HStack(spacing: 0) {
+                segment(title: firstOption)
+                    .accessibility(.segmentedSwitchFirstOption)
+                separator
+                segment(title: secondOption)
+                    .accessibility(.segmentedSwitchSecondOption)
+            }
+            .background(selectionBackground)
+            .background(roundedBackground)
+            .fixedSize(horizontal: false, vertical: true)
+        }
+        .accessibilityElement(children: .ignore)
+        .accessibility(label: .init(label))
+        .accessibility(value: .init(selection?.description ?? ""))
+        .accessibility(hint: .init(accessibilityHint))
+        .accessibility(addTraits: .isButton)
+        .accessibility(addTraits: selection != nil ? .isSelected : [])
+    }
+
+    @ViewBuilder func segment(title: String) -> some View {
+        Text(title)
+            .padding( .xSmall)
+            .frame(maxWidth: .infinity)
+            .contentShape(Rectangle())
+            .onTapGesture {
+                selectedIndex = title == firstOption ? 0 : 1
+            }
+    }
+
+    @ViewBuilder var separator: some View {
+        separatorColor
+          .frame(width: separatorWidth)
+    }
+
+    @ViewBuilder var selectionBackground: some View {
+        HStack {
+            if selection == secondOption {
+                Spacer(minLength: 0)
+                    .layoutPriority(1)
+            }
+
+            Color.whiteDarker
```

Comment:
For dark mode, the `.cloudDarkActive` might be a better choice for now

Reply:
Is it up to us? I mean whiteDarker has its own set for light and dark mode. Do we anywhere else do check `UITraitCollection.current.userInterfaceStyle == .dark`?

Reply:
We do, in cases where the color itself is not enough and we need to switch to completely different color for dark mode.